ArduCopter/system.cpp

```diff
@@ -288,12 +288,22 @@ void Copter::update_dynamic_notch()
 {
     const float ref_freq = ins.get_gyro_harmonic_notch_center_freq_hz();
     const float ref = ins.get_gyro_harmonic_notch_reference();
-    if (is_zero(ref)) {
-        ins.update_harmonic_notch_freq_hz(ref_freq);
-    }
-    else {
-        // calculate the harmonic notch filter frequency scaled on motor rpm
-        ins.update_harmonic_notch_freq_hz(ref_freq * MAX(1.0f, sqrtf(motors->get_throttle_out() / ref)));
+    if (is_tradheli()) {
+#if RPM_ENABLED == ENABLED
+        if (rpm_sensor.healthy(0) && !is_zero(ref)) {
+            ins.update_harmonic_notch_freq_hz(MAX(ref_freq, rpm_sensor.get_rpm(0) * ref / 60.0f));
+        } else {
+            ins.update_harmonic_notch_freq_hz(ref_freq);
+        }
+#endif
+    } else {
+        if (is_zero(ref)) {
```

Comment:
Not sure if we want to completely keep the tradheli handling separate or not but if we're happy to mix them up a bit with multicopters, we could have a line higher up that does the "if (is_zero(ref)) { ins.update_harmonic_notch_freq_hz(ref_freq); }" and then returns.  a non-functional change so it's up to you guys whether you think that's more understandable or less.

Reply:
the only issue with this suggestion is that it looks like for tradheli, if there's no RPM_ENABLED then this function is a no-op.

Reply:
@rmackay9 I agree with the non-functional change.

For tradheli without an RPM filter the notch becomes static at ref_freq which is I think what they want (that's how it is right now)